src/NUnitEngine/nunit.engine/Services/RuntimeFrameworkService.cs

```diff
@@ -50,26 +53,37 @@ public bool IsAvailable(string name)
 
         private static readonly Version AnyVersion = new Version(0, 0);
 
-        private static bool FrameworksMatch(RuntimeFramework f1, RuntimeFramework f2)
+        private static bool FrameworksMatch(RuntimeFramework requested, RuntimeFramework available)
         {
-            var rt1 = f1.Runtime;
-            var rt2 = f2.Runtime;
-
-            if (rt1 != RuntimeType.Any && rt2 != RuntimeType.Any && rt1 != rt2)
+            if (!RuntimesMatch(requested.Runtime, available.Runtime))
                 return false;
 
-            var v1 = f1.ClrVersion;
-            var v2 = f2.ClrVersion;
+            var requestedVersion = requested.FrameworkVersion;
+            var availableVersion = available.FrameworkVersion;
+
+            if (requestedVersion == AnyVersion)
+                return true;
+
+            return requestedVersion.Major == availableVersion.Major &&
+                   requestedVersion.Minor == availableVersion.Minor &&
+                   (requestedVersion.Build < 0 || availableVersion.Build < 0 || requestedVersion.Build == availableVersion.Build) &&
+                   (requestedVersion.Revision < 0 || availableVersion.Revision < 0 || requestedVersion.Revision == availableVersion.Revision) &&
+                   requestedVersion.Major == availableVersion.Major &&
+                   requestedVersion.Minor == availableVersion.Minor;
```

Comment:
```suggestion
                   (requestedVersion.Revision < 0 || availableVersion.Revision < 0 || requestedVersion.Revision == availableVersion.Revision);
```

If I'm reading right (I may well not be, in this class! 😂), I think the last two checks have been included twice here?